### DISCUSSION TITLE
Result.value.cacheType

### DISCUSSION THREAD

**Discussion by adamdahan** [onevcat/Kingfisher#1942] [Q&A] [answered]
Hi all, 

I am a little confused about the `ImagePrefetcher` and how it works in `.setImage`. I have some pre-determined URL's to grab images from so on app start I use the prefetcher like so: 
```
  prefetcher = ImagePrefetcher(urls: URLManager.shared.urls) { skippedResources, failedResources, completedResources in
            print("Skipped: \(skippedResources.count)")
            print("Failed: \(failedResources.count)")
            print("Completed: \(completedResources.count)")
  }
prefetcher.start()
```
The `Completed` resources count shows **100** and subsequently on new app launches `Skipped` shows **100** which is expected. But then in my `cellForRowAtIndexPath` I use this code (in which I log out the **result.value.cacheType:**  and it shows **none**. If the images are prefetched should the `cacheType` be memory or disk? 

```
let processor = DownsamplingImageProcessor(size: CGSize(width: 500, height: 500))
        cell.backgroundImageView.kf.indicatorType = .activity
        cell.backgroundImageView.kf.setImage(
            with: url,
            placeholder: UIImage(named: "placeholderImage"),
            options: [
                .processor(processor),
                .scaleFactor(UIScreen.main.scale),
                .transition(.fade(1))
            ])
        {
            result in
            switch result {
            case .success(let value):
                // From where the image was retrieved:
                  // - .none - Just downloaded.
                  // - .memory - Got from memory cache.
                  // - .disk - Got from disk cache.
                  print(value.cacheType)
            case .failure(let error):
                print("Job failed: \(error.localizedDescription)")
            }
   }
   ``` 
   PS: During my testing I am waiting for the `prefetcher` to complete before I scroll the collection of images.
   
   Any help to clarify what is going on here would be great! Thanks for the wonderful lib. @onevcat

**Comment by onevcat** [comment, marked as answer]
Kingfisher's image cache will consider the processor as a factor of cache policy. Images from different processors would be treated differently in the cache. Can you try to use the same processor (passing it to options, just like what you do in `setImage`) when you perform with `ImagePrefetcher`?

**Reply by adamdahan** [reply]
@onevcat I just tried to use the same **DownsamplingImageProcessor** and I pass in the option for `.cacheOriginalImage` and now it works as expected. 

Does this sound reasonable?

**Reply by onevcat** [reply]
Yes. If you can get `.memory` as the cache type in the result of `setImage` when the two options match, it just works.

**Reply by adamdahan** [reply]
Perfect. Thanks @onevcat !

**Reply by Punisherkh89** [reply]
О
